### PR TITLE
enables audio padding rather than cropping for HuBERT finetuning and decoding. Also fixes the issue that the decoding results are non-deterministic due to random audio cropping

### PR DESCRIPTION
**Pull request by freewym**
# Before submitting

- [ ] Was this discussed/approved via a Github issue? (no need for typos, doc improvements)
- [x] Did you read the [contributor guideline](https://github.com/pytorch/fairseq/blob/main/CONTRIBUTING.md)?
- [ ] Did you make sure to update the docs?
- [x] Did you write any new necessary tests?

## What does this PR do?
Enables audio padding rather than cropping for HuBERT finetuning and decoding (it shouldn't crop the audio during CTC finetuning, otherwise we may lose audio corresponding to some part of the transcript). Also fixes the issue that the decoding results of HuBERT  models are non-deterministic across different runs of inference with exactly the same setting, by disabling random_crop during inference (also disable it for the fine-tuning stage, as there seems no point doing wave crop for fine-tuning, which is also consistent with wav2vec2).

## PR review
Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

**Comment by freewym** [comment]
@dianaml0 Could you also please consider this fix?

**Comment by freewym** [comment]
@hirofumi0810 could you please do a review for this?
